dbms/tests/queries/0_stateless/00950_column_encoding_double_delta.sql

```diff
@@ -0,0 +1,211 @@
+DROP TABLE IF EXISTS reference;
+DROP TABLE IF EXISTS doubleDelta;
+
+CREATE TABLE reference (
+    key      UInt64,
+    valueU64 UInt64,
+    valueU32 UInt32,
+    valueU16 UInt16,
+    valueU8  UInt8,
+    valueI64 Int64,
+    valueI32 Int32,
+    valueI16 Int16,
+    valueI8  Int8,
+    valueDT  DateTime,
+    valueD   Date
+) Engine = MergeTree ORDER BY key;
+
+
+CREATE TABLE doubleDelta (
+    key      UInt64   CODEC(DoubleDelta),
+    valueU64 UInt64   CODEC(DoubleDelta),
+    valueU32 UInt32   CODEC(DoubleDelta),
+    valueU16 UInt16   CODEC(DoubleDelta),
+    valueU8  UInt8    CODEC(DoubleDelta),
+    valueI64 Int64    CODEC(DoubleDelta),
+    valueI32 Int32    CODEC(DoubleDelta),
+    valueI16 Int16    CODEC(DoubleDelta),
+    valueI8  Int8     CODEC(DoubleDelta),
+    valueDT  DateTime CODEC(DoubleDelta),
+    valueD   Date     CODEC(DoubleDelta)
+) Engine = MergeTree ORDER BY key;
+
+
+-- n^3 covers all double delta storage cases, from small difference between neighbour values (stride) to big.
+INSERT INTO reference (key, valueU64, valueU32, valueU16, valueU8, valueI64, valueI32, valueI16, valueI8, valueDT, valueD)
+	SELECT number as n, n * n * n as v, v, v, v, v, v, v, v, toDateTime(v), toDate(v) FROM system.numbers LIMIT 1, 100;
+
+-- best case - constant stride
+INSERT INTO reference (key, valueU64, valueU32, valueU16, valueU8, valueI64, valueI32, valueI16, valueI8, valueDT, valueD)
+	SELECT number as n, n as v, v, v, v, v, v, v, v, toDateTime(v), toDate(v) FROM system.numbers LIMIT 101, 100;
+
+-- checking for overflow
+INSERT INTO reference (key, valueU64, valueI64)
+VALUES (201, 18446744073709551616, 9223372036854775808), (202, 0, -9223372036854775808), (203, 18446744073709551616, 9223372036854775808);
+
+-- worst case - random stride
+INSERT INTO reference (key, valueU64, valueU32, valueU16, valueU8, valueI64, valueI32, valueI16, valueI8, valueDT, valueD)
+	SELECT number as n, n + (rand64() - 9223372036854775808)/1000 as v, v, v, v, v, v, v, v, toDateTime(v), toDate(v) FROM system.numbers LIMIT 301, 100;
+
+
+INSERT INTO doubleDelta SELECT * FROM reference;
+
+-- same number of rows
+SELECT a[1] - a[2] FROM (
+	SELECT groupArray(1) AS a FROM (
+		SELECT count() FROM reference
+		UNION ALL
+		SELECT count() FROM doubleDelta
+	)
+);
+
+SELECT 'U64';
+SELECT
+	key,
+	r.valueU64, d.valueU64, r.valueU64 - d.valueU64 as dU64
+FROM reference as r, doubleDelta as d
+WHERE
+	r.key == d.key
```

Comment:
It does not test if data in tables is the same but some values are mixed. It's better to avoid JOIN here and insert both raw and DeltaDelta values into one table and get results as
```
select * from t where i64 != i64_delta or i32 != i32_delta ...
```

Reply:
Done